### PR TITLE
components spanish translation

### PR DESCRIPTION
**Pull request by ybrodsky**

### CODE REVIEW

File: es/controllers/components.rst [hunk 1]
```diff
@@ -1,14 +1,288 @@
...
+    );
+
+El código anterior pasará un array que contiene precision y randomGenerator a 
+``MathComponent::__construct()`` como el segundo parámetro. Por convención 
+si los keys de un array son iguales a las propiedades públicas de un componente, 
+las propiedades seran actualizadas con los valores de esas keys.
+
+
+Usando otros componentes en tús componentes
+----------------------------------------
```
Comment (markstory) [inline es/controllers/components.rst]: Underlines need to match text length.

File: es/controllers/components.rst [hunk 2]
```diff
@@ -1,14 +1,288 @@
...
+    $this->OneTimer->getTime();
+
+.. note::
+
+    Tenga en cuenta que cargar un componente en tiempo de ejecución no 
+    llamara el método initialize de el componente. 
+
+
+Callbacks de componentes
+===================
```
Comment (markstory) [inline es/controllers/components.rst]: Underlines need to match text length.

File: es/controllers/components.rst [hunk 3]
```diff
@@ -1,14 +1,288 @@
...
 .. note::
-    La documentación no es compatible actualmente con el idioma español en esta página.
 
-    Por favor, siéntase libre de enviarnos un pull request en
-    `Github <https://github.com/cakephp/docs>`_ o utilizar el botón **Improve this Doc** para proponer directamente los cambios.
+    Usar un alias en un componente reemplaza esa instancia en cualquier lugar que ese componente sea 
+    usado, incluyendo dentro de otros componentes.
+
+Usando Componentes
+================
```
Comment (markstory) [inline es/controllers/components.rst]: Underlines need to match text length.

File: es/controllers/components.rst [hunk 4]
```diff
@@ -1,14 +1,288 @@
...
+
+Cada uno de los componentes principales esta detallado en su propio capítulo. 
+Ver :doc:`/core-libraries/toc-components`.
+Esta sección describe como configurar y usar los componentes y como crear tus propios 
+componentes.
+
+.. _configuring-components:
+
+Configurando componentes
+======================
```
Comment (markstory) [inline es/controllers/components.rst]: Underlines need to match text length.

File: es/controllers/components.rst [hunk 5]
```diff
@@ -1,14 +1,288 @@
...
+    }
+
+.. note::
+    En contraste con los componentes incluidos en un controlador 
+    ningún callback va a ser disparado en el componente de un componente.
+
+.. _component-api:
+
+API Componentes
+=============
```
Comment (markstory) [inline es/controllers/components.rst]: Underlines need to match text length.

File: es/controllers/components.rst [hunk 6]
```diff
@@ -1,14 +1,288 @@
...
+                    'action' => 'login'
+                )
+            ),
+            'Cookie' => array('name' => 'CookieMonster')
+        );
+
+El fragmento de código anterior sería un ejemplo de configuración de un componente 
+con el array ``$components``.
+Todos los componentes principales permiten que su configuración sea 
+seteada de ésta forma. A su vez, es posibile configurar el componente
```
Comment (antograssiot) [inline es/controllers/components.rst]: posibile -> posible

**Comment by markstory** [comment]
Thanks :+1:

**Pull request merged**